informer/src/informer_service/informer_service_handler.rs

```diff
@@ -98,3 +121,68 @@ impl InformerService {
         }
     }
 }
+
+pub struct BlockCreatedEventGenerator {
+    client: Arc<Client>,
+    rate_limiter: Limiter,
+    send_status: ColdEventSendStatus,
+}
+
+impl BlockCreatedEventGenerator {
+    pub fn new(client: Arc<Client>) -> Self {
+        Self {
+            client,
+            rate_limiter: ratelimit::Builder::new().quantum(1).interval(Duration::new(0, 5000)).build(),
+
+            send_status: ColdEventSendStatus::InProgress,
+        }
+    }
+
+    pub fn run(mut self, connection: Connection, mut favorite_block_number: u64) {
+        println!("I am here 3");
+        let rt = Runtime::new().unwrap();
+        rt.spawn(async move {
+            task::spawn(async move {
+                loop {
+                    let event = self.gen(favorite_block_number);
+                    connection.cold_notification(&event);
+                    self.rate_limiter.wait();
+                    match self.send_status {
+                        ColdEventSendStatus::AllEventsSent => {
+                            cinfo!(INFORMER, "Client has been informed about all events");
+                            let send_done =
+                                ColdEvents::SendingDone("You are informed about all you need".parse().unwrap());
+                            connection.cold_notification(&send_done);
+                            // FIXME : Cancel the subscription
+                            thread::sleep(Duration::from_secs(1_000));
+                        }
+                        ColdEventSendStatus::InProgress => {
+                            favorite_block_number += 1;
+                        }
+                    }
+                }
+            })
+        });
+        rt.shutdown_timeout(Duration::from_millis(100));
+    }
+}
+impl BlockCreatedEventGenerator {
+    fn gen(&mut self, favorite_block_number: u64) -> ColdEvents {
+        let best_block_number = self.client.best_block_header().number();
+        let mut event = ColdEvents::None;
+        if best_block_number >= favorite_block_number {
+            let current_id = BlockId::Number(favorite_block_number);
+            let block = self.client.block(&current_id).map(|block| {
+                let block = block.decode();
+                RPCBlock::from_core(block, self.client.network_id())
+            });
+            let current_block = block
+                .expect("In the process of generating block generation cold event response, the block is not created");
+            event = ColdEvents::BlockGeneration(Box::new(current_block));
+            if favorite_block_number == best_block_number {
+                self.send_status = ColdEventSendStatus::AllEventsSent;
```

Comment:
Changing self.send_status in the `gen` function is awkward. When you write a function, please consider whether this function can modify variables. When you make a function to modify the self's state, you should give a meaningful name or concept to the function.

When I read the function's name `gen` i don't expect the function to modify the self's state. Please do not modify any state in this function. Please read self's state, and return generated events.